testutil/integration/evmos/utils/staking.go

```diff
@@ -3,39 +3,14 @@
 package utils
 
 import (
-	"fmt"
 	"time"
 
 	errorsmod "cosmossdk.io/errors"
-	cryptotypes "github.com/cosmos/cosmos-sdk/crypto/types"
 	sdk "github.com/cosmos/cosmos-sdk/types"
-	stakingtypes "github.com/cosmos/cosmos-sdk/x/staking/types"
-	commonfactory "github.com/evmos/evmos/v16/testutil/integration/common/factory"
-	"github.com/evmos/evmos/v16/testutil/integration/evmos/factory"
 	"github.com/evmos/evmos/v16/testutil/integration/evmos/grpc"
 	"github.com/evmos/evmos/v16/testutil/integration/evmos/network"
 )
 
-func Delegate(tf factory.TxFactory, delegatorPriv cryptotypes.PrivKey, validatorAddr string, amount sdk.Coin) error {
-	delegatorAccAddr := sdk.AccAddress(delegatorPriv.PubKey().Address())
-
-	msgDelegate := stakingtypes.NewMsgDelegate(
-		delegatorAccAddr.String(),
-		validatorAddr,
-		amount,
-	)
-
-	resp, err := tf.ExecuteCosmosTx(delegatorPriv, commonfactory.CosmosTxArgs{
-		Msgs: []sdk.Msg{msgDelegate},
-	})
-
-	if resp.Code != 0 {
-		err = fmt.Errorf("received error code %d on Delegate transaction. Logs: %s", resp.Code, resp.Log)
-	}
-
-	return err
-}
-
```

Comment:
why was this moved to a `txFactory` method? In terms of abstraction this is better placed in the staking utilities. The factory interface should only expose the core primitives of building and signing transactions. cc @facs95 this is how I understood it?

Reply:
mm.. I think of it as a txFactory for any tx in a cosmos chain.. Similarly that the txFactory for evmos, where we have `DeployContract`, `CallContractAndCheckLogs`, etc..

Reply:
I originally also added more utils here, but @facs95 convinced me to keep this interface as lean as possible - maybe we need to align more on this? In any case we should have one common approach in the team 👍

Reply:
Yes at least in what I had in mind, I had initially thought of building the module specific msgs / transactions within the utils.. given that this will continuously increase so in this way we have the common building/braodcasting functions within tx factory and the module specific tooling for generating messages within the utils. This includes the more complex tasks such as waiting for proposals etc.. But happy to change my mind.. 

If the `Delegate` function is changed into, `CreateMsgDelegate` (within the utils), this would not need the factory in the signature and then the broadcasting happens through the factory as a second step. I kind of like those steps instead of having hidden logic. But happy to change my mind around this, just personal preferences.

Reply:
OK, refactored based on these comments